src/pull.go

```diff
@@ -28,6 +29,8 @@ func (f *PullFlags) Init(cmd *cobra.Command) {
 
 func (f *PullOnlyFlags) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.SourceURL, "source-url", "https://github.com", "The domain to pull from")
+	cmd.Flags().StringVar(&f.Token, "source-token", "", "The optional token to use to pull the repository")
```

Comment:
I think we'll need to add validation here based on the combination of flags. For example, a client could pass `source-user` but not `source-token` (and vice versa). A test for this behavior would be nice as well.

Reply:
added validation and relative unit test